src/System.Memory/tests/Memory/Trim.cs

```diff
@@ -0,0 +1,425 @@
+        }
+
+        [Fact]
+        public static void MemoryExtensions_TrimEnd_Single_Null()
+        {
+            var values = new Foo[] { null, null, 1, 2, null, null };
+            Foo trim = null;
+            var expected = new Foo[] { null, null, 1, 2 };
+
+            Memory<Foo> memory = new Memory<Foo>(values).TrimEnd(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, memory.ToArray()));
+
+            ReadOnlyMemory<Foo> rom = new ReadOnlyMemory<Foo>(values).TrimEnd(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, rom.ToArray()));
+
+            Span<Foo> span = new Span<Foo>(values).TrimEnd(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, span.ToArray()));
+
+            ReadOnlySpan<Foo> ros = new ReadOnlySpan<Foo>(values).TrimEnd(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, ros.ToArray()));
+        }
+
+        [Fact]
+        public static void MemoryExtensions_TrimEnd_Multi_Null()
+        {
+            var values = new Foo[] { null, 1, 2, 3, null, 2, 1, null };
+            var trim = new Foo[] { null, 1, 2 };
+            var expected = new Foo[] { null, 1, 2, 3 };
+
+            Memory<Foo> memory = new Memory<Foo>(values).TrimEnd(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, memory.ToArray()));
+
+            ReadOnlyMemory<Foo> rom = new ReadOnlyMemory<Foo>(values).TrimEnd(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, rom.ToArray()));
+
+            Span<Foo> span = new Span<Foo>(values).TrimEnd(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, span.ToArray()));
+
+            ReadOnlySpan<Foo> ros = new ReadOnlySpan<Foo>(values).TrimEnd(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, ros.ToArray()));
+        }
+
+        [Fact]
+        public static void MemoryExtensions_Trim_Single_Null()
+        {
+            var values = new Foo[] { null, null, 1, 2, null, null };
+            Foo trim = null;
+            var expected = new Foo[] { 1, 2 };
+
+            Memory<Foo> memory = new Memory<Foo>(values).Trim(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, memory.ToArray()));
+
+            ReadOnlyMemory<Foo> rom = new ReadOnlyMemory<Foo>(values).Trim(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, rom.ToArray()));
+
+            Span<Foo> span = new Span<Foo>(values).Trim(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, span.ToArray()));
+
+            ReadOnlySpan<Foo> ros = new ReadOnlySpan<Foo>(values).Trim(trim);
+            Assert.True(System.Linq.Enumerable.SequenceEqual(expected, ros.ToArray()));
+        }
+
+        [Fact]
+        public static void MemoryExtensions_Trim_Multi_Null()
```

Comment:
Do we have a test for trimming `default` span/memory?

For example:
```C#
Span<Foo> span = default;
default.Trim(trim);
default.Trim(default);
//etc.
```

Reply:
I think so - indirectly - latest commit has the following:
```
[InlineData(new int[] { }, new int[] { }, new int[] { })] // default.Trim(default [])
[InlineData(new int[] { }, new int[] { 1 }, new int[] { })] // default.Trim([1])
[InlineData(new int[] { 1 }, new int[] { }, new int[] { 1 })] // values.Trim(default [])
[InlineData(new int[] { 1 }, new int[] { 1 }, new int[] { })] // values.Trim([1])
public static void MemoryExtensions_TrimStart_Multi(int[] values, int[] trims, int[] expected)
{
    Memory<int> memory = new Memory<int>(values).TrimStart(trims);
    Assert.True(System.Linq.Enumerable.SequenceEqual(expected, memory.ToArray()));
```